presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -249,40 +245,23 @@ public static Slice stPolygon(@SqlType(VARCHAR) Slice input)
     @SqlType(DOUBLE)
     public static double stArea(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     {
-        OGCGeometry geometry = deserialize(input);
-
-        // The Esri geometry library does not support area for geometry collections. We compute the area
-        // of collections by summing the area of the individual components.
-        GeometryType type = GeometryType.getForEsriGeometryType(geometry.geometryType());
-        if (type == GeometryType.GEOMETRY_COLLECTION) {
-            double area = 0.0;
-            GeometryCursor cursor = geometry.getEsriGeometryCursor();
-            while (true) {
-                com.esri.core.geometry.Geometry esriGeometry = cursor.next();
-                if (esriGeometry == null) {
-                    return area;
-                }
-
-                area += esriGeometry.calculateArea2D();
-            }
-        }
-        return geometry.getEsriGeometry().calculateArea2D();
+        return deserialize(input).getArea();
     }
 
     @Description("Returns a Geometry type object from Well-Known Text representation (WKT)")
     @ScalarFunction("ST_GeometryFromText")
     @SqlType(GEOMETRY_TYPE_NAME)
     public static Slice stGeometryFromText(@SqlType(VARCHAR) Slice input)
     {
-        return JtsGeometrySerde.serialize(jtsGeometryFromWkt(input.toStringUtf8()));
+        return serialize(jtsGeometryFromWkt(input.toStringUtf8()));
     }
 
     @Description("Returns a Geometry type object from Well-Known Binary representation (WKB)")
     @ScalarFunction("ST_GeomFromBinary")
     @SqlType(GEOMETRY_TYPE_NAME)
     public static Slice stGeomFromBinary(@SqlType(VARBINARY) Slice input)
     {
-        return serialize(geomFromBinary(input));
+        return GeometrySerde.serialize(geomFromBinary(input));
```

Comment:
The `GeometrySerde.serialize` and `GeometrySerde.deserialize` are confusing. Should we rename `GeometrySerde` to `EsriGeometrySerde`?